app/src/main/java/com/example/dogbook/main/models/Post.java

```diff
@@ -78,26 +92,36 @@ public boolean isLikedByLoggedUser() {
         return isLikedByLoggedUser;
     }
 
-    public void setLikedByLoggedUser(boolean liked) {
-        isLikedByLoggedUser = liked;
-    }
-
     public int getLikesCount() {
         return likesCount;
     }
 
-    public void setLikesCount(int likesCount) {
-        this.likesCount = likesCount;
-    }
-
     public int getCommentsCount() {
         return commentsCount;
     }
 
+    public void setLikedByLoggedUser(boolean likedByLoggedUser) {
+        isLikedByLoggedUser = likedByLoggedUser;
+    }
+
+    public void setLikesCount(int likesCount) {
+        this.likesCount = likesCount;
+    }
+
     public void setCommentsCount(int commentsCount) {
         this.commentsCount = commentsCount;
     }
 
+    private void likedLocally() {
+        isLikedByLoggedUser = true;
+        likesCount +=1;
+    }
+
+    private void dislikedLocally() {
+        isLikedByLoggedUser = false;
+        likesCount -=1;
+    }
```

Comment:
We should code defensively here and only increment or decrement if we have that `isLikedByLoggedInUser` was in the expected state